label.js

```diff
@@ -19,7 +19,20 @@ async function label() {
   const context = github.context;
   const repoName = context.payload.repository.name;
   const ownerName = context.payload.repository.owner.login;
-  const issueNumber = context.payload.issue.number;
+  var issueNumber;
+
+  if (context.payload.issue !== undefined) {
+    issueNumber = context.payload.issue.number;
+  } else if (context.payload.pull_request !== undefined) {
+    issueNumber = context.payload.pull_request.number;
+  } else if (context.payload.project_card !== undefined) {
+    var issuesUrl = context.payload.project_card.issues_url;
+    var urlArray = issuesUrl.split("/");
+    issueNumber = urlArray[urlArray.length - 1];
+    if (issueNumber == "issues") {
```

Comment:
Sorry, it looks like this is trying to check that this `issue_url` does not end in `issues`. I was curious when that was, looking at the docs for the webhook event: https://developer.github.com/webhooks/event-payloads/#project_card it looks like it should have a `number` on the end of it?

Reply:
Yes, that was my intention. Because it's not every card that has a issue.

I didn't find a better way. Either on https://developer.github.com/webhooks/event-payloads/#project_card

or https://developer.github.com/v3/projects/cards/.

Looking at the object, the declaration `var issuesUrl = context.payload.project_card.issues_url;` may be wrong.

Maybe there's a better way?

Reply:
Looking at the object, the following declaration may be wrong:
```js
var issuesUrl = context.payload.project_card.issues_url;
```

There's some way to test it?